controlpanel/api/elasticsearch.py

```diff
@@ -59,7 +68,13 @@ def app_logs(app, num_hours=None):
 
     s = s.params(preserve_order=True)
 
-    for entry in s.scan():
-        entry['timestamp'] = parse(entry['@timestamp'])
-        yield entry
 
+    try:
+        logs = list(s.scan())
+        log.info("Got {} log entries.".format(len(logs)))
+        for entry in logs:
+            entry['timestamp'] = parse(entry['@timestamp'])
+    except Exception as ex:
```

Comment:
I'm assuming there isn't a more specific exception for when the request timeout 👌 .

Reply:
I tried the expected `ConnectionError` but that wasn't caught. :-/

Reply:
Yeah, I didn't find anything more specific neither among the exceptions on GH so probably this is the best we can do. 

I suppose it's the kind of additional information on the page that if can't be retrieved [because of an exception] it's not the end of the world and as I user I'd rather have the page mostly work, and I can always refresh to retry.

Reply:
I was going to say I could make the template output a helpful message if no logs are available. But the `app-log` macro calls directly to the API and I'm trying to work out why this is a good idea (i.e. I don't want to change something because I don't know the wider context).